docs/mkcloud.md

```diff
@@ -208,7 +228,7 @@ fi
 
 export cloudpv=${loused}
 export cloudsource=develcloud7
-export cloud=$cloudsource
+export cloud=${cloudsource:0:9}
```

Comment:
You don't like long names? :wink:

Reply:
This is there to allow the longer $cloudsource (e.g. develcloud7) to satisfy the conditions for $cloudbrlimit at line 131ff. in mkcloud.

Reply:
As you probably already figured out, `$cloud` is required to uniquely identify this cloud on this host, in case it is hosting multiple clouds.  So it is used for things like picking a name for `$cloudbr`, the cloud's network bridge, which is done by the code you referenced.

For the `mkch[a-p].cloud.suse.de` hosts in Nuremberg which are used for Jenkins CI and manual usage, `$cloud` is set to things like `va1` or `vp8` which are short and unique.  However when running your own mkcloud host if you rely on this line defaulting the value to something based on cloudsource, then truncating it like you have proposed will result in `develclou` which will cause a collision if you attempt to run two different clouds both with `cloudsource` set to `develcloud$something`.  And this might even cause problems even if the two clouds are not running simultaneously.  So I don't think this is the right approach.  (Even if it was, you have an off-by-two error since `${cloudsource:0:11}` should be sufficient, and anyway the truncation should be based on `$cloudbrlimit` not a hard-coded value.)

My suggestion would be to change the code to actually require `cloud` to be set manually, and if it isn't, to abort with an error explaining what the variable is for and that it should be set to something unique (at least, within this host).

Reply:
FYI - the mkcloud code is actually looking at $cloud plus + br, so that is why I truncated it to 9 characters. (See line 133 in mkcloud.) Having said that, I think just setting it to a static value is the best call here. Also, bear in mind that this is example code for a suggested mkcloud wrapper script. These changes are intended to give new developers something that will actually work for them out of the box. I am not sure what the rationale is for not including some of this stuff within mkcloud itself but this seems to be the status quo.